mantle/kola/harness.go

```diff
@@ -434,13 +434,15 @@ func filterTests(tests map[string]*register.Test, patterns []string, pltfrm stri
 
 		tagMatch := false
 		for _, tag := range Tags {
-			tagMatch = hasString(tag, t.Tags)
+			tagMatch = hasString(tag, t.Tags) || tag == t.RequiredTag
 			if tagMatch {
 				break
 			}
 		}
 
-		if t.RequiredTag != "" && !hasString(t.RequiredTag, Tags) {
+		// default to skipping tests with required tags unless the tag was given *or* a
+		// matching non-empty name pattern matches
+		if t.RequiredTag != "" && !tagMatch && (noPattern || !match) {
```

Comment:
no gonna lie this logic makes my head hurt. any way to break it down to make it easier to read? Maybe even commenting where some of these variables are defined would help.

Reply:
Yeah, it's really not great. The next one if-statement condition is pretty opaque too. I'll try to either tweak or at least add comments in a follow-up.

Reply:
https://github.com/coreos/coreos-assembler/pull/2595 should help.